functions/player/fn_addReplayPart.sqf

```diff
@@ -1,6 +1,22 @@
-params ["_array", "_index"];
+#include "script_component.hpp"
 
-// hintsilent format ["appending array %1", count GRAD_REPLAY_DATABASE_LOCAL];
-// diag_log format ["appending array %1", count GRAD_REPLAY_DATABASE_LOCAL];
+params ["_chunk", "_startIndex"];
 
-GRAD_REPLAY_DATABASE_LOCAL set [_index, (parseSimpleArray _array)];
\ No newline at end of file
+// add all received parts to database
+{
+    GRAD_REPLAY_DATABASE_LOCAL set [_startIndex + _forEachIndex, _x];
+} forEach _chunk;
```

Comment:
Comment in line 5 sounds like this is supposed to be a append?
```sqf
GRAD_REPLAY_DATABASE_LOCAL deleteRange [_startIndex, count _chunk];
GRAD_REPLAY_DATABASE_LOCAL append _chunk;
```
How about that? 2 lines of code instead of a loop

Reply:
As far as I understood the code that this replaces, this does not necessarily happen in order, so `GRAD_REPLAY_DATABASE_LOCAL` might be `[12,12,nil,nil,12,12]` and you are trying to set indices 2 and 3.

Reply:
yes, iirc i did this to prevent chunks being played in the wrong order, as we cant guarantee they arrive in order (multiple remoteexecs)